src/halmos/sevm.py

```diff
@@ -2311,15 +2312,39 @@ def finalize(ex: Exec):
                             ex.path.append(codesize > 0)
                     ex.st.push(codesize)
 
-                # TODO: define f_extcodehash for known addresses in advance
+                elif opcode == EVM.EXTCODECOPY:
+                    account_addr = self.resolve_address_alias(ex, uint160(ex.st.pop()))
+                    account_code: Contract = (
+                        ex.code.get(account_addr, None) or ByteVec()
+                    )
```

Comment:
I see, how should we handle that?

Reply:
i don't have a good idea, but for now, we could either:

1. print a warning that the address is assumed to be new, having no code in it

2. raise a halmos exception

Reply:
let's print a warning and hope it only happens rarely

Reply:
do you have a sense for the kind of pattern that would cause the solver in `resolve_address_alias` to time out?

Reply:
You can try running the farcaster test after reducing the timeout for the address resolving, and will see the complex term. If i remember correctly, it’s involved with several bit manipulations. It's not very complicated per se, but the timeout for resolution is just 1 second. Increasing it may hurt performance, when there are frequent calls or sends to symbolic addresses.

Reply:
yeah, let's try printing a warning and reevaluate it later.

Reply:
also it seems like this could be unsound if there are multiple addresses that it may alias: we would return an empty bytevec, but it seems like we should instead instantiate new paths

e.g. imagine `account` may alias `A` and `B`, in this PR we would have:
path 1 => extcodecopy(account) is empty

I think instead we should do this:

path 1, condition `account == A`, extcodecopy(account) returns code[A]
path 2, condition `account == B`, extcodecopy(account) returns code[B]
path 3, condition `account != A and account != B`, extcodecopy(account) is empty

wdyt?

Reply:
yes, a sound approach in general is to check `account == x` for every existing address `x`, and then branch a new path for each of them, unless the check result is not unsat. this captures what you described.

my concern was this may lead to path explosion. in the worse case, `N` paths are newly created for each opcode that requires address alias resolution, where N is the number of known addresses. that's why we don't support multiple aliases for now.

but perhaps it's time to experiment the more general approach, at least to see how bad the path explosion would be in practice.